### PR TITLE
Replace legacy Allowed-Origins with Origins-Pattern on all distros

### PR DESCRIPTION
**Pull request by towo**
#### Pull Request (PR) description

Since all Debian-based distributions VP officially supports feature a version of `unattended-upgrade` that's compatible with the `Origins-Pattern` variable, this PR gets rid of the legacy configuration altogether.

To ensure that no one gets unexpected surprises by this, I've added input validation for `$origins` to ensure it's only in the new-style format and hopefully avoid exploding in people's faces; nevertheless, since this potentially breaks existing configurations, it should probably be combined with a major version bump.

### CODE REVIEW

**Review by kenyon — APPROVED** [review]

**Comment by kenyon** [comment]
Looks good, maybe a more descriptive PR title for the changelog?

File: types/origin.pp [hunk 1]
```diff
@@ -0,0 +1 @@
+type Unattended_upgrades::Origin = Pattern[/^(origin|codename|label|site|suite|component|archive|[oalcn])=[^,]+(,(origin|codename|label|site|suite|component|archive|[oalcn])=[^,]+)*/]
```
Comment (bastelfreak) [inline types/origin.pp:1]: please add a puppet-strings documentation:
```
# @summary this does type does blah...
```

File: metadata.json [hunk 2]
```diff
@@ -27,7 +27,8 @@
       "operatingsystem": "Ubuntu",
       "operatingsystemrelease": [
         "18.04",
-        "20.04"
+        "20.04",
+        "21.04"
```
Comment (bastelfreak) [inline metadata.json:31]: do you have a ubuntu 21.04 box around to provide factsets for https://github.com/voxpupuli/facterdb ? Also please create a new PR that adds 21.04 here. That allows us to put the two PRs in different sections (breaking change vs enhancement)

**Comment by towo** [comment]
> Looks good, maybe a more descriptive PR title for the changelog?

To be fair… I didn't even notice I forgot to change the PR title from the branch name. ta. ;)

**Comment by nox-4O4** [comment]
You might consider updating the README. Currently documentation there still references the now-removed `legacy_origin`, along with the explanation what it used to change.

**Comment by kenyon** [comment]
Thanks, I made #205 for updating the README.

**Pull request merged**